test/Marketplace.test.js

```diff
@@ -1344,4 +1424,45 @@ describe("Marketplace", function () {
       expect(await marketplace.mySlots()).to.not.contain(slotId(slot))
     })
   })
+
+  describe("protocol fee", function () {
+    const dead = "0x000000000000000000000000000000000000dead"
+
+    it("is burned when storage request is created", async function () {
+      switchAccount(client)
+      let priceWithProtocolFee = maxPriceWithProtocolFee(request, config)
+      await token.approve(marketplace.address, priceWithProtocolFee)
+      const startBalance = await token.balanceOf(client.address)
+      const startDeadBalance = await token.balanceOf(dead)
+      await expect(marketplace.requestStorage(request)).to.emit(
+        marketplace,
+        "StorageRequested"
+      )
+      const endBalance = await token.balanceOf(client.address)
+      const endDeadBalance = await token.balanceOf(dead)
+      expect(startBalance - endBalance).to.equal(
+        maxPriceWithProtocolFee(request, config)
+      )
+      expect(endDeadBalance - startDeadBalance).to.equal(
+        protocolFee(request, config)
+      )
+    })
+
+    it("is not returned when request is cancelled", async function () {
```

Comment:
Maybe we should add a similar test for failed? Since now I believe with the Vault the client gets refunded if the request fails...